UIP-0027.md

```diff
@@ -0,0 +1,88 @@
+# UIP-27: Transaction service fields
+
+```
+Author:   Dima Saveliev <dima@thirdhash.com>
+Status:   Draft
+Created:  2019-05-07
+```
+
+## Abstract
+
+  This document describes protocol changes in the transaction structure:
+the `nVersion` field is going to be replaced by four separate service fields.
+
+
+## Motivation
+
+  UIP-0018 introduced a transaction type concept and the corresponding field was added
+as part of already existing `nVersion` field. That was achieved via some chunks
+of bit arithmetic and was a kind of ad-hoc hack.
+
+  By implementing separate service fields we solve the following problems:
+  
+  1. Clean-up the logic behind the `type` and `version` fields.
+  2. Minimize the size of `type` and `version` fields.
+  3. Preserve compatibility with Bitcoin codebase.
+  
+  As an additional benefit, we will have two more fields for possible future needs.
+  
+  
+## Specification
+
+At the given moment we have the next `nVersion` field structure:
+
+```
+  0x00      0x00     0x00      0x00   
+^------^  ^------^  ^----------------^
+ unused     type         version      
+^------------------------------------^
+             nVersion
+```
+
+* `nVersion` - original `uint32_t` value
+* `type` - derived `uint8_t` attribute
+* `version` - derived `uint16_t` attribute
+
+Proposed layout for the same bytes:
+```
+  0x00      0x00      0x00      0x00     
+^------^  ^------^  ^------^  ^------^   
+reserved    type    reserved  version    
+```
+
+* `type` - independent`uint8_t` field
+* `version` - independent`uint8_t` field
+
+
+## Rationale
+
+The rationale behind this format is quite simple because in fact, we don't have a choice.
+It's very hard to add, remove or even swap existing bytes due to a couple of reasons:
+
+* Tests contain a lot of magic numbers, based on the assumptions regarding transaction size.
+* Some tests heavily rely on hardcoded, unique transactions from the real Bitcoin network.
+* Unit-e requires a regular syncing with Bitcoin codebase.
```

Comment:
Regular syncing is not _required_. It's better to say that we want to regularly sync with Bitcoin Core.

Reply:
Agree with this!